libtcmu.c

```diff
@@ -591,7 +591,7 @@ static int open_devices(struct tcmulib_context *ctx)
 
 	for (i = 0; i < num_devs; i++) {
 		char *tmp_path;
-		char buf[256] = {'\0'};
+		char buf[CFGSTRING_MAX] = {'\0'};
```

Comment:
We get the cfgstring in those 2 functions from a sysfs file so it is limited to a page (4k on most archs). I think that is the only tcmu/tcmu-runner restriction. I think LIO is going to be limited by a page from its configfs use.

Could you move the the common file open/read related parts of open_devices and is_uio to a helper and call it. We will just have the one place to worry about it for the file reading case then.

For the max cfgstring name,  would it be easier to just strdup whatever we get from the kernel and find in the sysfs file? runner at least never has to worry about it. For the max sysfs file size, we could just do stat on "/sys/class/uio/%s/name" to get the initial max size for the read buffer.